app/services/archive_uploader.rb

```diff
@@ -0,0 +1,69 @@
+class ArchiveUploader
+  # see: https://docs.ovh.com/fr/storage/pcs/capabilities-and-limitations/#max_file_size-5368709122-5gb
+  # officialy it's 5Gb. but let's avoid to reach the exact spot of the limit
+  MAX_FILE_SIZE_FOR_BACKEND_BEFORE_CHUNKING = ENV.fetch('ACTIVE_STORAGE_FILE_SIZE_THRESHOLD_BEFORE_CUSTOM_UPLOAD') { 4.gigabytes }
+
+  def upload
+    return upload_with_active_storage if active_storage_service_local?
+
+    case File.size(filepath)
+    when 0..MAX_FILE_SIZE_FOR_BACKEND_BEFORE_CHUNKING then
+      upload_with_active_storage
+    else
+      upload_with_custom_wrapper
+    end
+  end
+
+  def active_storage_service_local?
+    Rails.application.config.active_storage.service == :local
+  end
+
+  private
+
+  attr_reader :procedure, :archive, :filepath
+
+  def upload_with_active_storage
+    archive.file.attach(
+      io: File.open(filepath),
+      filename: object_filename,
+      metadata: { virus_scan_result: ActiveStorage::VirusScanner::SAFE }
+    )
+  end
+
+  def upload_with_custom_wrapper
+    blob = ActiveStorage::Blob.create_before_direct_upload!(
+        key: SecureRandom.uuid,
+        filename: object_filename,
+        byte_size: File.size(filepath),
+        checksum: Digest::SHA256.file(filepath).hexdigest,
+        content_type: 'application/zip',
+        metadata: { virus_scan_result: ActiveStorage::VirusScanner::SAFE }
+      )
+
+    if syscall_to_custom_uploader
+      archive.file.purge if archive.file.attached?
+      ActiveStorage::Attachment.create(
+        name: 'file',
+        record_type: 'Archive',
+        record_id: archive.id,
+        blob_id: blob.id
+      )
+    else
+      fail "should be retried ?"
+    end
+  end
+
+  def object_filename
+    archive.filename(procedure)
+  end
+
+  def syscall_to_custom_uploader
+    system ENV.fetch('ACTIVE_STORAGE_BIG_FILE_UPLOADER_WITH_ENCRYPTION_PATH').to_s, filepath, blob.key
```

Comment:
Ajout de parenthèse et du paramètre exception pour savoir quand il y a un soucis (https://rubyapi.org/3.1/o/kernel#method-i-system)

```suggestion
    system(ENV.fetch('ACTIVE_STORAGE_BIG_FILE_UPLOADER_WITH_ENCRYPTION_PATH').to_s, filepath, blob.key, exception: true)
```